File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:

    
        // int Operate(int a, int b, string token){
        //       if(token == "+")
        //           return a+b;
            
        //        if(token == "-")
        //           return a-b;

        //     if(token == "*")
        //         return (long)a * (long)b;
            
        //     if(token == "/")
        //        return a/b;

        //        return -1;
        // }


    int evalRPN(vector<string>& tokens) {
          stack<int> st;

          unordered_map<string,function<int (int,int)>>mp = {
               {"+",[](int a, int b){return a+b;}},
               {"-",[](int a, int b){return a-b;}}, 
               {"*",[](int a, int b){return (long)a*(long)b;}},
               {"/",[](int a, int b){return a/b;}}
          };

          for(string &token : tokens){
            if(token == "+" || token == "-" || token == "*" || token == "/"){
                  
                  int b = st.top();
                  st.pop();

                  int a = st.top();
                  st.pop();

                //   int result = Operate(a,b,token);
                  int result = mp[token](a,b);


                  st.push(result);
            }else{
                st.push(stoi(token));
            }
          }  

          return st.top();

           
       

    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
    
    let stack = [];

    for(let token of tokens){
        if(["+","-","*","/"].includes(token)){
            let b = stack.pop();
            let a = stack.pop();

            let result;

            if(token === "+") result = a+b;
            else if(token === "-") result = a-b;
            else if(token === "*") result = a*b;
            else if(token === "/") result = Math.trunc(a/b);

            stack.push(result);
        }else {
            stack.push(parseInt(token,10));
        }
    }

    return stack.pop();
};